apps/iframe/src/index.tsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react'
+import ReactDOM from 'react-dom'
+import { ConnectService } from './services/connectService'
+import { UIProvider } from '@3id/ui-provider'
+//@ts-ignore
+import * as hexToRgb from 'hex-to-rgb'
+
+import './index.css'
+import App from './Components/App/App'
+// TODO: readd this here
+// import reportWebVitals from './reportWebVitals'
+
+import type { UIProviderHandlers } from '@3id/ui-provider'
+
+/**
+ *  UI Window Functions
+ */
+
+const mobileRegex = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i
+const checkIsMobile = () => mobileRegex.test(navigator.userAgent)
+
+const error = (error: any) => `
+  <p class='walletSelect_error'>${error}</p>
+`
+
+// Given a request will render UI module templates
+const render = async (params: object, type: string, functions: { accept: any; decline: any }) => {
+  const request = Object.assign({}, params, { type })
+
+  // TODO:
+
+  //TODO: parse through data object to get colors
+  ReactDOM.render(
+    <React.StrictMode>
+      <App
+        renderType={render}
+        bkgColor={hexToRBGA('#000000', 10)}
+        acceptFunction={functions.accept}
+        declineFunction={functions.decline}
+      />
+    </React.StrictMode>,
+    document.getElementById('root')
+  )
+
+  // document.getElementById('root').innerHTML = template({ request }, checkIsMobile())
+}
+
+/**
+ *  Identity Wallet Service configuration and start
+ */
+
+const connectService = new ConnectService()
+
+const modalView = async (params: object, type: string) => {
```

Comment:
This logic should probably be handled in the React tree directly rather than at the top-level and re-rendering the entire React tree, so we can leverage React for state management.